downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,422 @@
+
+- Modified the behavior of the Semantic API `typeDescriptor()` methods to return the intersection type symbol instead of the effective type symbol (i.e., the symbol of the corresponding non-intersection type that is computed to represent the same value space).
+    ```ballerina
+    // `TypeSymbol` of `A` will now be an `IntersectionTypeSymbol` instead of `ArrayTypeSymbol`.
+    type A int[] & readonly;
+    ```
+
+- Removed the support for the `AES/GCM/PKCS5Padding` encryption algorithm from the `crypto` package.
+
+## Platform updates
+
+### New features
+
+#### Introduction of list constants
+
+The language now supports list constants. You can declare a constant variable using a list constructor as shown below.
+
+```ballerina
+const int[] ERROR_CODES = [404, 500, 503];
+const ERROR_MESSAGES = ["Not Found", "Internal Server Error", "Service Unavailable"];
+```
+
+Also, you can declare a list constant using a byte array literal.
+
+```ballerina
+const byte[] data16 = base16 `55 EE 66`;
+const data64 = base64 `ABCD pqrs`;
+```
+
+### Improvements
+
+#### Make `async-send` an action
+
+
+Reclassified `async-send` from a statement to an action in alignment with the Ballerina specification.
+
+The following case is now supported.
+
+```ballerina
+public function main() {
+    worker w1 {
+        _ = 5 -> w2;
+    }
+}
+```
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.8.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3ATeam%2FCompilerFE+milestone%3A2201.8.0+is%3Aclosed+label%3AType%2FBug).
+
+## Runtime updates
+
+### New features
+
+#### New Runtime Java APIs
+
+Introduced the `Type getImpliedType(Type)` API in the `io.ballerina.runtime.api.utils.TypeUtils` class to recursively resolve type reference types (to get referred types) and/or intersection types (to get effective types).
+
+    ```ballerina
+    // `getImpliedType` on type A returns a `BArrayType`. This is achieved by first retrieving the referred type of the type reference type, which will be an intersection type, and then retrieving the effective type of the intersection type.
+    type A int[] & readonly;
+    ```
+
+#### Ballerina Profiler (experimental)
+
+Introduced the `profile` CLI command, which runs a Ballerina package and does a CPU profile of it during runtime.
+
+```
+$ bal profile
+```
+
+- For example, if we run the above command in the root directory of a Ballerina package, it generates a flame graph that shows the time taken to execute each function.
+
+- The output is given by the `ProfilerOutput.html` file, which can be opened using a web browser.
+
+>**Note:** This is an experimental feature, which supports only a limited set of functionality.
+
+### Improvements
+
+#### Java 17 support
+
+Ballerina now supports code compilation and execution with Java 17.
```

Comment:
We better have more details on why, impact to the user etc.

Reply:
@warunalakshitha

Reply:
We have two sections mentioned about Java 17 support. Shall we add one sections under runtime improvement.

As an example

Java 17 support
The Java 11 Long term Support (LTS) active support will end in September 2023. Therefore, the jBallerina runtime is now upgraded to support [Java 17 LTS](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html), which is the latest long-term support release of the Java SE platform.

Users will not have an impact on the Java version upgrade except for the following.

If you are specifically using Java 11 to run Ballerina programs, you would have to switch to Java 17.

Ballerina interoperability implementation will have an impact on Java 17 support due to any incompatible changes. As an example, Java 17 will have some restrictions on using Java reflections with internal Java packages.

For more information, see the [Java 17 release notes](https://www.oracle.com/java/technologies/javase/17-relnote-issues.html).

Reply:
Also is it ok to use oracle release note as the link https://www.oracle.com/java/technologies/javase/17-relnote-issues.html @anupama-pathirage . Anyway I couldn't found any other proper release note.

Reply:
I think we still have to mention it in two places.

1. in Backward-incompatible changes section - There we can have only a summary but better link to the more detailed description which is available under runtime section.
2. in the improvements section of runtime udpates- This is the detailed version. 

So +1 to have the detailed version suggested by @warunalakshitha above under the improvements section of runtime updates

Reply:
> Also is it ok to use oracle release note as the link https://www.oracle.com/java/technologies/javase/17-relnote-issues.html @anupama-pathirage . Anyway I couldn't found any other proper release note.


+1

Reply:
Updated.

Reply:
In the backward incompatibility section writup link to the improvements section of runtime updates instead to the https://github.com/ballerina-platform/ballerina-dev-website/blob/85e25044962b792fca8f6e33b4e180592d87a383/downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md#java-17-support

Reply:
Why should we point to the `Improvements` section rather than the specific `Java 17 support` section?

Reply:
yes, that is what i meant by improvements section in high-level (specifically toJava 17 support title)

Reply:
Sorry my bad, seems it is already pointed to the correct link.